fl4health/clients/fed_pca_client.py

```diff
@@ -0,0 +1,143 @@
+import random
+import string
+from logging import INFO
+from pathlib import Path
+from typing import Dict, Tuple, Type, TypeVar
+
+import torch
+from flwr.client.numpy_client import NumPyClient
+from flwr.common import Config, NDArrays, Scalar
+from flwr.common.logger import log
+from torch import Tensor
+from torch.utils.data import DataLoader
+
+from fl4health.model_bases.pca import PCAModule
+from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+
+T = TypeVar("T")
+
+
+class FedPCAClient(NumPyClient):
```

Comment:
Why did you decide to use NumPyClient instead of basic_client?

Reply:
Because `BasicClient` assumes we are performing mini-batch training with a PyTorch model, which I thought was too distinct from how federated PCA works. I could use `BasicClient` instead, but I would still need to overwrite most of its methods